bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/FutureValue.java

```diff
@@ -72,4 +75,18 @@ public void stamp(BType type, List<TypeValuePair> unresolvedValues) {
      public Object copy(Map<Object, Object> refs) {
          return null;
      }
+
+     public boolean isDone() {
+         return isDone;
+     }
+
+     public boolean isCancelled() {
+         return isCancelled;
+     }
+
+     public boolean cancel() {
+         this.strand.cancel = true;
```

Comment:
why we need the strand also to have the cancel flag aprt from FutureValue having it?

Reply:
FutureValue is the reference to a strand that others can use it to cancel the strand. Within functions, strand is the one that carry this meta info. We need this info within functions. A strand cannot access it's own future. Hence this cancel flag is used to track that there is a cancel requested.